unlock-app/src/components/interface/locks/Settings/forms/hooksComponents/GitcoinContractHook.tsx

```diff
@@ -0,0 +1,84 @@
+import { CustomComponentProps } from '../UpdateHooksForm'
+import { Button, Input } from '@unlock-protocol/ui'
+import { useEffect } from 'react'
+import { useFormContext } from 'react-hook-form'
+import {
+  useGetLockSettings,
+  useSaveLockSettings,
+} from '~/hooks/useLockSettings'
+
+export const GitcoinContractHook = ({
+  lockAddress,
+  network,
+  setEventsHooksMutation,
+}: CustomComponentProps) => {
+  const { isLoading: isLoadingSettings, data: settings } = useGetLockSettings({
+    lockAddress,
+    network,
+  })
+
+  const { mutateAsync: saveSettingsMutation } = useSaveLockSettings()
+
+  const { handleSubmit, register, getValues, setValue } = useFormContext()
+
+  // initialize the form field with the fetched setting
+  useEffect(() => {
+    if (settings?.requiredGitcoinPassportScore !== undefined) {
+      setValue(
+        'hook.requiredGitcoinPassportScore',
+        settings.requiredGitcoinPassportScore
+      )
+    }
+  }, [settings, setValue])
+
+  const onSubmit = async (values: any) => {
+    const requiredGitcoinPassportScore = getValues(
+      'requiredGitcoinPassportScore'
+    )
+
+    await saveSettingsMutation({
+      lockAddress,
+      network,
+      requiredGitcoinPassportScore: requiredGitcoinPassportScore,
+    })
+
+    await setEventsHooksMutation.mutateAsync(values)
+  }
+
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+      className="flex flex-col gap-2 text-sm"
+    >
+      <p>
+        With this hook, you can control membership purchases exclusively for
+        members with a required Gitcoin passport score.
+      </p>
+
+      <Input
+        size="small"
+        description="Required Gitcoin Passport Score"
+        type="number"
+        {...register('hook.requiredGitcoinPassportScore', {
+          valueAsNumber: true,
+          min: 0,
```

Comment:
```suggestion
          min: 15,
```
I wanted to learn a bit more about the score and they have [this in the docs](https://support.gitcoin.co/gitcoin-knowledge-base/gitcoin-passport/common-questions/how-is-gitcoin-passports-score-calculated)

> The threshold for Gitcoin Passport Scoring is 15 and the maximum possible score is 100. Any score above 15 will be eligible for matching funding for their donations.